File: Models/Models/SearchResult.cs
using Models.Enums;
using Models.Models;
using Rado.Models.Authentication;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Rado.Models
{
    public class SearchResult
    {
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public UserCount userCount { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public UserView userView { get; set; }
        public DisplayPartView[] data { get; set; }
        public long duration { get; set; }
        public int size { get; set; }
        public long currentItem { get; set; }
        public int clientId { get; set; }
        public Filter filter { get; set; }
    }
}

File: Rado/part365/src/app/model/searchResult.ts
import { DisplayPartView } from './displayPartView'
import { Filter } from './filters/filter'
import { FilterRimWithTyre } from './FilterRimWithTyre'
import { UserView } from './userView'
import { UserCount } from './userCount'

export class SearchResult {
    data?: DisplayPartView[]
    duration?: number
    currentItem?: number
    clientId?: number

    size?: number
    filter?: Filter
    filterRimWithTyre?: FilterRimWithTyre
    userCount?: UserCount
    userView?: UserView
}
